core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -290,7 +291,7 @@ public String toString() {
     this.metadataFileLocation = metadataFileLocation;
     this.formatVersion = formatVersion;
     this.uuid = uuid;
-    this.location = location;
+    this.location = location != null ? LocationUtil.stripTrailingSlash(location) : null;
```

Comment:
Nit: I feel like we should just change `LocationUtil.stripTrailingSlash` to return null if a null location is passed in instead of throwing an NPE otherwise I think we have to do this null check everywhere where we don't have a location defined in advance. But it's a public utility which may be used beyond the Iceberg library itself so I understand not wanting to change it at this point.

Reply:
I think this is the only place location could be null? In all other places location is already initialized and must be non-null.

Reply:
I see if that's the case then we should just leave as is imo